pkg/generate/route.go

```diff
@@ -59,7 +59,7 @@ func Route(cr *regopapi.ImageRegistry, route *regopapi.ImageRegistryConfigRoute,
 			},
 		},
 	}
-	if cr.Spec.TLS {
+	if len(route.SecretName) > 0 {
```

Comment:
as discussed this morning, i think this was correct as is.  If the backend registry is using TLS, the route should be "re-encrypt" and if the backend registry is not using TLS, the route should be "Edge termination".

Whether the route is given a certificate or not is up to the user (whether they want to supply one or not).

Reply:
well maybe not correct as is, but the logic should be:

1) if backend TLS, create reencrypt route
2) if backend !TLS, create edge termination route
3) either way, if user provided a route.SecretName, add TLS certificates to the route